### ISSUE TITLE
Sometimes PSI result is wrong

### ISSUE THREAD

**Issue by stone-wlg** [Visa-Research/volepsi#7]
Env:
- Centos 7.9
- sender.csv / 315MB / 10,000,000
- receiver.csv / 315MB / 10,000,000
- sender.csv == receiver.csv, (id is 32 characters, ex: xxxxxxxxxxxxxxxxxxxxxx1000000000)

Steps:
- Loop 100 times
- One by one

Results:
- 6/100 psi result is 9,999,999 lines
- 94/100 psi result is 10,000,000 lines

Log:
=> 1 sender starting
=> 1 receiver starting
=> 1 psi done
=> 1 psi result: 9999999
=> 2 sender starting
=> 2 receiver starting
=> 2 psi done
=> 2 psi result: 9999999
=> 3 sender starting
=> 3 receiver starting
=> 3 psi done
=> 3 psi result: 10000000
=> 4 sender starting
=> 4 receiver starting
=> 4 psi done
=> 4 psi result: 10000000
=> 5 sender starting
=> 5 receiver starting
=> 5 psi done
=> 5 psi result: 10000000
...

addtion:
- 1,000,000 psi is 100% correct
- 100,000,000 psi occurs wrong results, the same as 10,000,000 psi

**Comment by ladnir** [comment]
Thanks for the details. If you run the same input twice do you get the same answer?

**Comment by ladnir** [comment]
Was able to reproduce. Will fix soon

**Comment by ladnir** [comment]
Should be fixed now

**Comment by stone-wlg** [comment]
> Should be fixed now

ok, let me try again.

**Comment by stone-wlg** [comment]
still be able to reproduce.

**Comment by ladnir** [comment]
And you are sure you're on the latest commit?

**Comment by ladnir** [comment]
can you run ` -u Baxos_solve_Test -n 10000000 -b 16384 -t 1000 -v` and tell me if it fails

**Comment by RindalVisa** [comment]
closing due to inactivity.
